clients/privacy-center/components/consent/NoticeDrivenConsent.tsx

```diff
@@ -171,7 +171,7 @@ const NoticeDrivenConsent = () => {
           (n) => n.privacy_notice_history_id === historyKey
         );
         const servedNotice = servedNotices?.find(
-          (sn) => sn.privacy_notice_history.id === historyKey
+          (sn) => sn.privacy_notice_history?.id === historyKey
```

Comment:
this file is the only one with "actual" changes. there were two API changes we account for here:
1. served notices now return optional `privacy_notice_history` (since TCF doesn't have those)
2. the preferences patch now returns the data we expect nested under `preferences` (so that tcf data can be nested in its own subsections)

every other change in this PR is pretty much type wrangling

Reply:
Nice, it looks like a lot of good type updates on the privacy center side made it into this PR

Reply:
yes, the privacy center types were a bit out of date 😓